dll/cpl/hotplug/enum.c

```diff
@@ -4,4 +4,308 @@
  * FILE:        dll/cpl/hotplug/enum.c
  * PURPOSE:     device enumeration
  * PROGRAMMERS: Johannes Anderwald (johannes.anderwald@reactos.org)
+ *              Thamatip Chitpong (thamatip.chitpong@reactos.org)
  */
+
+#include "hotplug.h"
+
+#include <initguid.h>
+#include <devguid.h>
+
+#define MAX_DEVICE_DISPLAYNAME_LEN 40
+
+static
+HTREEITEM
+InsertDeviceTreeItem(
+    _In_ HTREEITEM hParent,
+    _In_ DEVINST DevInst,
+    _In_ PHOTPLUG_DATA pHotplugData)
+{
+    WCHAR szDisplayName[MAX_DEVICE_DISPLAYNAME_LEN];
+    WCHAR szGuidString[MAX_GUID_STRING_LEN];
+    TVINSERTSTRUCTW tvItem;
+    GUID ClassGuid;
+    INT nClassImage;
+    DWORD dwSize;
+    CONFIGRET cr;
+
+    /* Get the device description */
+    dwSize = sizeof(szDisplayName);
+    cr = CM_Get_DevNode_Registry_Property(DevInst,
+                                          CM_DRP_DEVICEDESC,
+                                          NULL,
+                                          szDisplayName,
+                                          &dwSize,
+                                          0);
+    if (cr != CR_SUCCESS)
+    {
+        LoadStringW(hApplet,
+                    IDS_UNKNOWN_DEVICE,
+                    szDisplayName,
+                    ARRAYSIZE(szDisplayName));
+    }
+
+    /* Get the class GUID */
+    dwSize = sizeof(szGuidString);
+    cr = CM_Get_DevNode_Registry_Property(DevInst,
+                                          CM_DRP_CLASSGUID,
+                                          NULL,
+                                          szGuidString,
+                                          &dwSize,
+                                          0);
+    if (cr == CR_SUCCESS)
+    {
+        pSetupGuidFromString(szGuidString, &ClassGuid);
+    }
+    else
+    {
+        memcpy(&ClassGuid, &GUID_DEVCLASS_UNKNOWN, sizeof(GUID));
```

Comment:
Wouldn't `ClassGuid = GUID_DEVCLASS_UNKNOWN;` work ?

Reply:
It should work, according to the code here:
[https://git.reactos.org/?p=reactos.git;a=blob;f=dll/win32/devmgr/devmgmt/DeviceNode.cpp;h=bcafd87c98627a8767b568f54dc9769f4b5933e7;hb=HEAD#l125](https://git.reactos.org/?p=reactos.git;a=blob;f=dll/win32/devmgr/devmgmt/DeviceNode.cpp;h=bcafd87c98627a8767b568f54dc9769f4b5933e7;hb=HEAD#l125)